test/e2e/util_gatewayapi_test.go

```diff
@@ -38,53 +37,27 @@ const (
 
 	// openshiftOperatorsNamespace holds the expected OSSM subscription and Istio operator pod.
 	openshiftOperatorsNamespace = "openshift-operators"
-	// openshiftIstioOperatorDeploymentName holds the expected istio-operator deployment name.
-	openshiftIstioOperatorDeploymentName = "istio-operator"
+	// openshiftIstioOperatorDeploymentName holds the expected Service Mesh
+	// operator deployment name.
+	openshiftIstioOperatorDeploymentName = "servicemesh-operator3"
 	// openshiftIstiodDeploymentName holds the expected istiod deployment name
 	openshiftIstiodDeploymentName = "istiod-openshift-gateway"
-	// openshiftSMCPName holds the expected OSSM ServiceMeshControlPlane name
-	openshiftSMCPName = "openshift-gateway"
+	// openshiftIstioName holds the expected Istio CR name.
+	openshiftIstioName = "openshift-gateway"
 	// cvoNamespace is the namespace of cluster version operator (CVO).
 	cvoNamespace = "openshift-cluster-version"
 	// cvoDeploymentName is the name of cluster version operator's deployment.
 	cvoDeploymentName = "cluster-version-operator"
 )
 
-// updateIngressOperatorRole updates the ingress-operator cluster role with cluster-admin privilege.
-// TODO - Remove this function after https://issues.redhat.com/browse/OSSM-3508 is fixed.
-func updateIngressOperatorRole(t *testing.T) error {
-	t.Helper()
-
-	// Create the same rolebinding that the `oc adm policy add-cluster-role-to-user` command creates.
-	// Caller must remove this after setting it.
-	crb := &rbacv1.ClusterRoleBinding{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "cluster-admin-e2e",
-		},
-		RoleRef:  rbacv1.RoleRef{APIGroup: rbacv1.GroupName, Kind: "ClusterRole", Name: "cluster-admin"},
-		Subjects: []rbacv1.Subject{{Kind: rbacv1.ServiceAccountKind, Name: "ingress-operator", Namespace: operatorcontroller.DefaultOperatorNamespace}},
-	}
-
-	// Add the rolebinding to the ingress-operator user.
-	if err := kclient.Create(context.TODO(), crb); err != nil {
-		if kerrors.IsAlreadyExists(err) {
-			t.Logf("rolebinding already exists")
-			return nil
-		}
-		t.Logf("error adding rolebinding: %v", err)
-		return err
-	}
-	t.Log("rolebinding has been added")
-	return nil
-}
-
-// assertCrdExists checks if the CRD of the given name exists and returns an error if not.
-// Otherwise returns the CRD version.
-func assertCrdExists(t *testing.T, crdname string) (string, error) {
+// assertCRDExists checks if the CRD of the given name exists and returns an
+// error if not.  If the does CRD exist, this function returns a slice strings
```

Comment:
```suggestion
// error if not.  If the CRD does exist, this function returns a slice of strings
```

Reply:
Thanks!  Fixed in https://github.com/openshift/cluster-ingress-operator/compare/30b576c4293b2962b137bcb93ff77f278d7bf681..26a3918c7374cc16eac04b87c9e4267591c304a5... and https://github.com/openshift/cluster-ingress-operator/compare/26a3918c7374cc16eac04b87c9e4267591c304a5..f9e552d6aeb31b42bd5791923de4b66f29ec107e (I fixed the "does", then noticed and fixed the "of"..